src/routes/spotify/desktop/Attract.svelte

```diff
@@ -27,6 +29,29 @@
       message = messages[(i + 1) % messages.length];
     }, 5000);
   }
+
+  const shuffleArray = <T,>(array: T[]): T[] =>
+    array
+      .map((value) => ({ value, sort: Math.random() }))
+      .sort((a, b) => a.sort - b.sort)
+      .map(({ value }) => value);
+
+  let albums1: AlbumTracks[] = [];
+  let albums2: AlbumTracks[] = [];
+  const shuffleAlbums = () => {
+    const shuffled = shuffleArray([...playlist.albums]);
+    const half = Math.ceil(shuffled.length / 2);
+    albums1 = shuffled.slice(0, half);
+    albums2 = shuffled.slice(half);
+  };
+
+  onMount(() => shuffleAlbums());
```

Comment:
nitpick but `onMount` and `if (browser)` both effectively do the same thing. `onMount` is the convention in the rest of the codebase so lets move the setInterval timer into that?

Reply:
I imagine something like this

```ts
import {chunk, shuffle} from "$lib/array";

let art0: string[] = [];
let art1: string[] = [];

const reshuffle = () => {
  let all = shuffle(playlist.albums.map((a) => a.art))
  let chunks = chunk(all, all.length / 2)
  art0 = chunks[0]
  art1 = chunks[1]
}

onMount(() => {
    setInterval(() => {
      const i = messages.findIndex((m) => m == message);
      message = messages[(i + 1) % messages.length];
    }, 5000);

  reshuffle()
});

$effect(() => {
  visible || reshuffle()
});
```

```svelte
```

Reply:
niiice, updated :+1: